tensorrt_llm/_torch/memory/gpu_memory_backend.py

```
@@ -0,0 +1,304 @@
+    @property
+    def is_rw(self) -> Optional[bool]:
+        return self._is_rw
+
+    def has_committed_weights(self) -> bool:
+        if self._client is None:
+            return False
+        try:
+            from gpu_memory_service.common.locks import GrantedLockType
+
+            return self._client.granted_lock_type == GrantedLockType.RO
+        except Exception:
```

Ruff (0.15.11)

[warning] Do not catch blind exception: `Exception`

(BLE001)